### PR TITLE
Alerting: add new notifier lark/feishu (#29360)

### LINKED ISSUE #29360: Alerting: add new notifier lark/feishu
**What would you like to be added**:
One new notifier for Lark/Feishu
**Why is this needed**:
[Lark](https://www.larksuite.com/) or [Feishu](https://www.feishu.cn/) is collaboration suite
developed by bytedance and supports bots in chat groups.

### PR DESCRIPTION
**Pull request by tomowang**
**What this PR does / why we need it**:
This PR add new notifier lark/feishu
**Which issue(s) this PR fixes**:

Fixes #29360

**Special notes for your reviewer**:
For testing whether alert was sent to Lark/Feishu chat group, the Lark/Feishu app must be installed.

### CODE REVIEW

File: docs/sources/alerting/notifications.md [hunk 1]
```diff
@@ -183,6 +184,23 @@ In DingTalk PC Client:
...
+
+1. Select one chat group. Click "Settings" icon on right of the panel.
+
+1. Click "BOTs" tab, then click "Add Bot" at the bottom of the panel.
+
+1. In "Add Bot" panel, select "Custom Bot". You can customize "Display Name" and "Description" of the bot. Click "Add".
+
+1. There will be a webhook URL in the panel, looks like this: https://open.larksuite.com/open-apis/bot/v2/hook/xxxxxxxxx or https://open.feishu.cn/open-apis/bot/hook/xxxxxxxxx. Copy this URL to the Grafana Lark/Feishu setting page and then click "Finish".
+
+Lark/Feishu support the following "message types": `text`, `post`, `share_chat`, `image` and `interactive`. Only the `text` message type is supported currently.
```
Comment (achatterjee-grafana) [inline docs/sources/alerting/notifications.md:202]:
```suggestion
[Instructions in Chinese](https://www.feishu.cn/hc/zh-cn/articles/360024984973-%E5%9C%A8%E7%BE%A4%E8%81%8A%E4%B8%AD%E4%BD%BF%E7%94%A8%E6%9C%BA%E5%99%A8%E4%BA%BA) / 
[Instructions in English](https://www.larksuite.com/hc/en-US/articles/360048487736-Use-Bots-in-group-chat)

In Lark/Feishu PC Client:

1. Select one chat group, then click the "Settings" icon on the right of the panel.

1. Click the "BOTs" tab, then click "Add Bot" at the bottom of the panel.

1. In the "Add Bot" panel, select "Custom Bot". You can customize "Display Name" and "Description" of the bot. 

1. Click "Add".

There will be a webhook URL in the panel, looks like this: https://open.larksuite.com/open-apis/bot/v2/hook/xxxxxxxxx or https://open.feishu.cn/open-apis/bot/hook/xxxxxxxxx. 

1. Copy this URL to the Grafana Lark/Feishu setting page and then click "Finish".

Lark/Feishu supports the following "message types": `text`, `post`, `share_chat`, `image` and `interactive`. Only the `text` message type is supported currently.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added some copy-edit suggestions.

**Comment by aknuds1** [comment]
Thank you for contributing to Grafana; at this point however, we won't add more notifiers to Grafana's core, unless they are highly requested. We are currently working on supporting backend plugins in #6841, once that has been released it should be fairly easy to build a plugin.

Again, thank you for contributing, even though this change didn't make it this time.

**Comment by tomowang** [comment]
Understand. Supporting backend plugins is a better choice.
We can achieve this alert channel by webhook and FaaS. So this is not a blocking issue.

**Comment by jujunchen** [comment]
I also want this feature

**Comment by AllanChain** [comment]
@jujunchen Maybe you need wait for #16004 to be implemented. Before that, you can just set up something like a webhook relay/translator.

**Comment by Songxwn** [comment]
Hope to support it soon

**Comment by AllanChain** [comment]
@Songxwn Feishu notifier won't be added to Grafana, see https://github.com/grafana/grafana/pull/29361#issuecomment-734170767. You can just set up a webhook relay like https://github.com/AllanChain/grafana-feishu, or try out new [custom JSON payload and template functions](https://grafana.com/docs/grafana/latest/alerting/configure-notifications/manage-contact-points/integrations/webhook-notifier/#custom-payload).

**Pull request closed without merging**